src/index.ts

```diff
@@ -123,6 +146,15 @@ export class StorageProvider {
         throw customError;
       }
     }
+
+    // Fetch chain id from provider
+    let network = await this.provider.getNetwork();
+    this.chainId = network.chainId;
+
+    // throw error if chain id is not equal to the chain id of the app
+    if (this.chainId !== chainId) {
+      throw new Error('Wrong network, please change the network to the arcana');
```

Comment:
'Wrong network. Change the network to Arcana'

Reply:
Done